ci/githubClient.sc

```diff
@@ -0,0 +1,170 @@
+    .timeout(connTimeoutMs = 5000, readTimeoutMs = 100000)
+    .postData(body)
+    .asString
+    .throwError
+}
+
+/**
+ * Comment with msg on pull request with pullNumber.
+ */
+def comment(pullNumber: String, msg: String, event: String = "COMMENT"): Unit = {
+  val request = Js.Obj(
+    "body"  -> Js.Str(msg),
+    "event" -> Js.Str(event)
+    )
+  val path = s"repos/mesosphere/marathon/pulls/$pullNumber/reviews"
+  execute(path, request.toString)
+}
+
+/**
+ * Reject pull request with pullNumber.
+ */
+def reject(
+  pullNumber: String,
+  buildUrl: String,
+  buildTag: String): Unit = {
+  val msg = s"I'm building your change at [$buildTag]($buildUrl)."
+
+  comment(pullNumber, msg, "REQUEST_CHANGES")
+}
+
+/**
+ * Collect test results.
+ *
+ * @return the parsed test results.
+ */
+@main
+def collectTestResults(): Js.Arr = {
+
+  try {
+    // Join all results
+    val testResults = ls! pwd / 'target / "phabricator-test-reports" |? ( _.ext == "json")
+    val joinedTestResults: Js.Arr = testResults.view.map(read!)
+      .map(upickle.json.read)
+      .collect { case a: Js.Arr => a }
+      .reduce { (l: Js.Arr, r: Js.Arr) =>
+        val n = l.arr ++ r.arr
+        Js.Arr(n :_*)
+      }
+
+    joinedTestResults
+  } catch {
+    case NonFatal(e) =>
+      utils.printlnWithColor(s"Could not collect test results: ${e.getMessage}", utils.Colors.BrightRed)
+      Js.Arr()
+  }
+}
+
+/**
+ * Report success of diff build back to GitHub.
+ *
+ * @param pullNumber The pull request of the build.
+ * @param buildUrl A link back to the build on Jenkins.
+ * @param buildTag Identifies build.
+ * @param maybeArtifact A description of the Marathon binary that has been uploaded.
+ *    It's None when now package was uploaded.
+ */
+def reportSuccess(
+  pullNumber: String,
+  buildUrl: String,
+  buildTag: String,
+  maybeArtifact: Option[awsClient.Artifact]): Unit = {
+
+  val testResults = collectTestResults()
+
+  // Collect unsound, i.e. canceled, tests
+  val unsoundTests = testResults.value
+    .collect { case test: Js.Obj if test("result").value == "unsound" => test  }
+  val hasUnsoundTests = unsoundTests.nonEmpty
+
+  // Construct message
+  val buildinfoDiff = maybeArtifact.fold(""){ artifact =>
+    s"""
+      |```json
+      |   "url": "${artifact.downloadUrl}",
+      |   "sha1": "${artifact.sha1}"
+      |```
+     """.stripMargin
+  }
+
+  var msg = s"""
+    |**\u2714 Build of #$pullNumber completed successfully.**
+    |
+    |See details at [$buildTag]($buildUrl).
+    |
+    |You can create a DC/OS with your patched Marathon by creating a new pull
+    |request with the following changes in [buildinfo.json](https://github.com/dcos/dcos/blob/master/packages/marathon/buildinfo.json):
+    |
+    |$buildinfoDiff
+    |
+    |""".stripMargin
+
+  if (!hasUnsoundTests) {
+    msg += "**＼\\ ٩( ᐛ )و /／**"
+  } else {
+    val unsoundTestsList: String = unsoundTests.foldLeft("") { (msg:String, test: Js.Obj) =>
```

Comment:
Somehow this is not working currently even though the tests report a skipped test. We are not using the `KnownIssue` flag anyways. So I'd say we go forward without this extra comment.

Reply:
https://jira.mesosphere.com/browse/MARATHON-7781